pkg/littlefs2/fs/littlefs2_fs.c

```diff
@@ -84,7 +84,11 @@ static int _dev_write(const struct lfs_config *c, lfs_block_t block,
           (void *)c, block, off, buffer, size);
 
     uint32_t page = (fs->base_addr + block) * fs->sectors_per_block * mtd->pages_per_sector;
-    return mtd_write_page_raw(mtd, buffer, page, off, size);
+    int ret = mtd_write_page_raw(mtd, buffer, page, off, size);
+    if (ret == -EIO) {
+        ret = LFS_ERR_CORRUPT;
```

Comment:
Since `EIO` is already used by the driver for bus errors, better use something different for errors detected by the device. How about `EFAULT`, `EBADMSG` or `EILSEQ`?

Reply:
This decision was made by the original author, I'll have to look up which return value is most suitable instead of EIO.

Reply:
From the MAN page of write:
```
       EIO    A low-level I/O error occurred while modifying the inode.
              This error may relate to the write-back of data written by
              an earlier write(), which may have been issued to a
              different file descriptor on the same file.
```

EFAULT seems to be related to the input data and not to the write process:
```
       EFAULT buf is outside your accessible address space.
```

For the other ones, only generic information is given in the errno documentation, but I don't think they fit better than EIO.
https://man7.org/linux/man-pages/man3/errno.3.html
```
EBADMSG Bad message (POSIX.1-2001)

EILSEQ Invalid or incomplete multibyte or wide character
              (POSIX.1, C99).

              The text shown here is the glibc error description; in
              POSIX.1, this error is described as "Illegal byte
              sequence".
```

However one thing I found that might fit if EIO is not good, would be EHWPOISON:
```
       EHWPOISON
              Memory page has hardware error.
```

This has the disadvantage though that "EHWPOISON" does not seem to be defined for the nRF52 and possibly for a lot of other platforms neither...
IMO the EIO error fits at this point.